pkg/solana/logpoller/log_poller.go

```diff
@@ -44,15 +68,94 @@ func New(lggr logger.SugaredLogger, orm ORM) *LogPoller {
 		Start: lp.start,
 	}.NewServiceEngine(lggr)
 	lp.lggr = lp.eng.SugaredLogger
+
 	return lp
 }
 
 func (lp *LogPoller) start(context.Context) error {
+	if lp.typeProvider == nil {
+		return ErrMissingEventTypeProvider
+	}
+	cl, err := lp.client.Get()
+	if err != nil {
+		return err
+	}
+	lp.collector = NewEncodedLogCollector(cl, lp, lp.lggr)
+
 	lp.eng.Go(lp.run)
 	lp.eng.Go(lp.backgroundWorkerRun)
 	return nil
 }
 
+func makeLogIndex(txIndex int, txLogIndex uint) int64 {
+	if txIndex < 0 || txIndex > math.MaxUint32 || txLogIndex > math.MaxUint32 {
+		panic(fmt.Sprintf("txIndex or txLogIndex out of range: txIndex=%d, txLogIndex=%d", txIndex, txLogIndex))
+	}
+	return int64(math.MaxUint32*uint32(txIndex) + uint32(txLogIndex))
+}
+
+// Process - process stream of events coming from log ingester
+func (lp *LogPoller) Process(programEvent ProgramEvent) (err error) {
```

Comment:
Loader does not provide a retry mechanic for Process. If Process fails - we'll skip logs

Reply:
A retry is built into the loader. If an error is returned here, the loader should retry using exponential backoff.